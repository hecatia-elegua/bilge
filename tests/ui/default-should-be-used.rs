#![feature(const_trait_impl)] use bilge::prelude::*;

#[bitsize(2)]
#[derive(DefaultBits)]
enum One {
    Small,
    #[default]
    Step,
}

#[bitsize(2)]
#[derive(DefaultBits)]
struct A {
    b: Inner,
}

#[bitsize(2)]
#[derive(FromBits)]
enum Inner {
    Zero,
    One,
    Two,
    Three,
}

fn main() {}
