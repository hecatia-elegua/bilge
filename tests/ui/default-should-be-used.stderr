error: use derive(Default) for enums
 --> tests/ui/default-should-be-used.rs:4:10
  |
4 | #[derive(DefaultBits)]
  |          ^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `DefaultBits` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find attribute `default` in this scope
 --> tests/ui/default-should-be-used.rs:7:7
  |
7 |     #[default]
  |       ^^^^^^^
  |
help: consider adding a derive
  |
3 + #[derive(Default)]
4 + #[bitsize(2)]
  |

error[E0277]: the trait bound `Inner: Default` is not satisfied
  --> tests/ui/default-should-be-used.rs:12:10
   |
12 | #[derive(DefaultBits)]
   |          ^^^^^^^^^^^ the trait `Default` is not implemented for `Inner`
   |
   = note: this error originates in the derive macro `DefaultBits` (in Nightly builds, run with -Z macro-backtrace for more info)
